pkg/api/v1/helper/helpers.go

```diff
@@ -513,3 +519,70 @@ func StorageNodeAffinityToAlphaAnnotation(annotations map[string]string, affinit
 	annotations[v1.AlphaStorageNodeAffinityAnnotation] = string(json)
 	return nil
 }
+
+// ReorganizeTaints returns the updated set of taints, taking into account old taints that were not updated,
+// old taints that were updated, old taints that were deleted, and new taints.
+func ReorganizeTaints(node *v1.Node, overwrite bool, taintsToAdd []v1.Taint, taintsToRemove []v1.Taint) (string, []v1.Taint, error) {
```

Comment:
Thanks @ravisantoshgudimetla for the explain, I was thinking put all of the functions to `taint util` maybe better as I can get all of the util fns for taint in one place and do not need to switch to other places. 

Let us wait what other's point for this issue.